### PR TITLE
[CI] Run ESIMD tests on Arc in precommit when ESIMD changed

### PR DESCRIPTION
**Pull request by sarnex**
Let's see if this works

### CODE REVIEW

**Comment by sarnex** [comment]
@aelovikov-intel Looks like ESIMD + Matrix tests on DG2 take ~14 min. Is that too long?

Let's ignore the failures for now, I don't know what's going on but if we want to proceed with this change I'll investigate.

**Comment by aelovikov-intel** [comment]
We've had about 75 pre-commit jobs in the last 24 hours. Just pre-commit would take about 18 hours at that pace, and we also use that single runner for post-commit/nightly.

I'd say yes, too long.

**Comment by sarnex** [comment]
@aelovikov-intel If we only ran it when there were ESIMD changes is that still too long? I'm not sure how many PRs we have per day but probably 1-3. Might I'll actually have to use REST.

**Comment by aelovikov-intel** [comment]
> @aelovikov-intel If we only ran it when there were ESIMD changes is that still too long? I'm not sure how many PRs we have per day but probably 1-3. Maybe I'll actually have to use REST.

In that case, it would be the right thing to do, even if that is too long. :)

I'm also wondering how much the two timed-out tests affected total time. Maybe if we can fix that, the overall time would drop as well.

**Comment by sarnex** [comment]
Good idea, let me try disabling those just for testing and see if it's still long. If so, I'll try to implement the code owner idea.

**Comment by sarnex** [comment]
@aelovikov-intel Mind reviewing? I think it's ready. In the current run the new runner didn't run because there were no ESIMD changes. There was a previous run [here](https://github.com/intel/llvm/actions/runs/7834829696/job/21379047671) and the only fail is unrelated and failing in HEAD.

File: .github/workflows/sycl-detect-changes.yml
Comment (aelovikov-intel) [inline .github/workflows/sycl-detect-changes.yml:60]:
```suggestion
            esimd:
```

to align with convention in this file.

File: .github/workflows/sycl-linux-precommit.yml
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-precommit.yml:134]: I'd prefer this merged into the matrix job at line 50. Not sure how to do it best.

**Comment by sarnex** [comment]
Will address feedback when I'm back from vacation!

**Comment by sarnex** [comment]
@aelovikov-intel Do you mind taking another look at this? I attempted to address your feedback, but I'm not sure if I did it in the cleanest way.

File: sycl/test-e2e/ESIMD/lit.local.cfg
Comment (sarnex) [inline sycl/test-e2e/ESIMD/lit.local.cfg:10]: We need this to fix failures when run on OCL. The current postcommit job only runs L0 anyway, so there's no difference in coverage. We should investigate why OCL fails separately.

Comment (aelovikov-intel) [reply sycl/test-e2e/ESIMD/lit.local.cfg:10]: That comment should go into the source file :)

**Review by aelovikov-intel — DISMISSED** [review]

**Comment by sarnex** [comment]
@intel/dpcpp-esimd-reviewers Please review this one ASAP, we can have DG2 testing in precommit after this is merged

File: .github/workflows/sycl-detect-changes.yml
Comment (aelovikov-intel) [inline .github/workflows/sycl-detect-changes.yml:100]: Hm, I've just realised that it's not in the precommit yml file. What was the reason? We haven't had any real logic here before this change.

Comment (sarnex) [reply .github/workflows/sycl-detect-changes.yml:100]: I felt it's related more to processing the changes list which we already do in that yml, and the complexity of the logic seems to be similar to the existing 'Set output' job in the same file. I can move it if you'd prefer but I'd ask to like to do it in a separate PR because my team needs this testing ASAP.

Comment (aelovikov-intel) [reply .github/workflows/sycl-detect-changes.yml:100]: Yes, I'd like it moved. The reason is that I want this file to detect undisputable facts only and the tests we want to run on a given runner is "subjective" in nature. I'm ok with doing it in the next PR.

**Review by aelovikov-intel — APPROVED** [review]

File: .github/workflows/sycl-detect-changes.yml
Comment (sarnex) [reply .github/workflows/sycl-detect-changes.yml:100]: Thanks I will do it immediate after this is merged

**Comment by sarnex** [comment]
Failures not related, see
https://github.com/intel/llvm/pull/12816
and
https://github.com/intel/llvm/actions/runs/8022089783/job/21917005858

**Pull request merged**
